js/src/client.ts

```diff
@@ -2676,113 +2682,182 @@ export class Client implements LangSmithTracingClientInterface {
     return result["examples"] as ExampleSearch[];
   }
 
+  public async createExample(update: ExampleCreate): Promise<Example>;
+
+  /**
+   * @deprecated This signature is deprecated, use createExample(update: ExampleCreate) instead
+   */
   public async createExample(
     inputs: KVMap,
     outputs: KVMap,
-    {
-      datasetId,
-      datasetName,
-      createdAt,
-      exampleId,
-      metadata,
-      split,
-      sourceRunId,
-    }: CreateExampleOptions
+    options: CreateExampleOptions
+  ): Promise<Example>;
+
+  public async createExample(
+    inputsOrUpdate: KVMap | ExampleCreate,
+    outputs?: KVMap,
+    options?: CreateExampleOptions
   ): Promise<Example> {
-    let datasetId_ = datasetId;
-    if (datasetId_ === undefined && datasetName === undefined) {
+    if ("dataset_id" in inputsOrUpdate || "dataset_name" in inputsOrUpdate) {
```

Comment:
Simplify all this with a user-defined type guard:

https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

You will be able to avoid the cast below